sql/updates/latest-dev.sql

```diff
@@ -0,0 +1,48 @@
+-- For continuous aggregates: Copy ACL privileges (grants) from the
+-- query view (user-facing object) to the internal objects (e.g.,
+-- materialized hypertable, direct, and partial views). We want to
+-- maintain the abstraction that a continuous aggregates is similar to
+-- a materialized view (which is one object), so privileges on the
+-- user-facing object should apply also to the internal objects that
+-- implement the continuous aggregate. Having the right permissions on
+-- internal objects is necessary for the watermark function used by
+-- real-time aggregation since it queries the materialized hypertable
+-- directly.
+DO $$
+DECLARE
+	rels regclass[];
+	rel regclass;
+	acl aclitem[];
+BEGIN
```

Comment:
My SQL-fu is probably not good enough, so if you have an example of how to do this it would be appreciated. 

I think the tricky part is how to run the update on multiple relations in a single query (the second loop with unnest).

Reply:
I'm actually not entirely sure if you need to use unnest at all.

AIUI, you want to update the chunk `relacl` with the hypertable `relacl` if that is not NULL, so if we simplify it slightly and say that you want to update the child `relacl` with the `relacl` of the parent for all tables, this would work:

```
WITH rels AS (SELECT inhrelid::regclass AS relid, relacl
              FROM pg_inherits JOIN pg_class ON oid = inhparent
             WHERE (SELECT relacl FROM pg_class WHERE oid = inhrelid) IS NULL
               AND relacl IS NOT NULL)
UPDATE pg_class SET relacl = rels.relacl FROM rels WHERE oid = rels.relid;
```

If you want to limit it to only specific tables (e.g., only hypertables), you have to add further restrictions, of course.

Note that this will overwrite the relacl, not update it.

Reply:
From what I can tell, your example isn't doing the same things. It doesn't include the view objects. I think I will stick with what is already implemented.

Reply:
No, that is correct, it does not include the view objects. It was just an example of how to handle it for the hypertables. It is done similarly for the direct view and the partial view, so that would be three updates. I'm fine with keeping this code, so go ahead and merge.

Reply:
I think I managed to figure something out. Can you please take another look?

Reply:
Looks good.